src/Controls/src/Core/Platform/Android/BottomNavigationViewUtils.cs

```diff
@@ -94,16 +94,27 @@ internal static void UpdateEnabled(bool tabEnabled, IMenuItem menuItem)
 				else
 				{
 					menuItem = menu.GetItem(i);
-					SetMenuItemTitle(menuItem, item.title);
-					loadTasks.Add(SetMenuItemIcon(menuItem, item.icon, mauiContext));
+					if (menuItem.ItemId != i)
+					{
+						menu.RemoveItem(menuItem.ItemId);
+						loadTasks.Add(SetupMenuItem(item, menu, i, currentIndex, bottomView, mauiContext, out menuItem));
+					}
+					else
+					{
+						SetMenuItemTitle(menuItem, item.title);
+						loadTasks.Add(SetMenuItemIcon(menuItem, item.icon, mauiContext));
+					}
 				}
 
 				menuItems.Add(menuItem);
 			}
 
-			if (showMore)
+			var menuSize = menu.Size();
+			if (showMore && menu.GetItem(menuSize - 1).ItemId != MoreTabId)
 			{
 				var moreString = context.Resources.GetText(Resource.String.overflow_tab_title);
+				if (menuSize == maxBottomItems)
```

Comment:
Do we still need these two lines of code?

Reply:
I have a crash without them :/

Reply:
yea, ok, I guess I could see some scenarios where that might happen

Like, if you have 5 items and then add a 6th
It'll need to transition that 5th item to a more tab